### PR TITLE
feature: allow sync flag to be passed to ThreadStream for synchronous…

### PR DESCRIPTION
**Pull request by karankraina**
Allow `sync` flag to be passed to ThreadStream for sync logging.
Ref - #1801

### CODE REVIEW

**Comment by karankraina** [comment]
@mcollina Do you think this works ?

**Review by mcollina — CHANGES_REQUESTED** [review]
Thanks for opening a PR! Can you please add a unit test?

**Comment by karankraina** [comment]
@mcollina Added unit test.

**Review by mcollina — APPROVED** [review]
lgtm

File: test/transport/syncTrue.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  })
+  const instance = pino(transport)
+
+  var value = { message: 'sync' }
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
```
Comment (mcollina) [inline test/transport/syncTrue.test.js:29]: I think you could do `instance.flushSync()` and then just try reading the file. It should work.

Comment (karankraina) [reply test/transport/syncTrue.test.js:29]: Wouldn't calling `flushSync` force synchronously flush all messages to the stream ? That would pass even if `sync` is `false` ? I tried to simulate a event loop block immediately after calling `instance.info`. I think I need to update the loop duration by some bit. Am i missing something?

Comment (mcollina) [reply test/transport/syncTrue.test.js:29]: ah no prob. This is ok

**Comment by mcollina** [comment]
You might need to pass `flush: true` to https://nodejs.org/api/fs.html#filehandlecreatewritestreamoptions in https://github.com/pinojs/pino/blob/f220965c3e3885b8479c37ec880e43315e1a2373/test/fixtures/to-file-transport.js#L8. Otherwise Node.js would not `fsync` the file descriptor, so you might not be able to read it.

**Comment by karankraina** [comment]
@mcollina Ah yess, updated PR. `flush: true` fixed it.  Added loop duration by a bit just in case.

File: test/transport/syncTrue.test.js (hunk 1 above)
Comment (karankraina) [reply test/transport/syncTrue.test.js:29]: Updated PR. Should pass now :)

**Comment by karankraina** [comment]
> @mcollina Ah yess, updated PR. `flush: true` fixed it. Added loop duration by a bit just in case.

Ubuntu 22 failing for some reason. :( Need to re-check this. Had worked on my fork

File: test/transport/syncTrue.test.js (hunk 1 above)
Comment (karankraina) [reply test/transport/syncTrue.test.js:29]: @mcollina can you retrigger the CI actions please

**Pull request merged**